### PR TITLE
Add test cases for JSON resource marshaling

### PR DESCRIPTION
**Pull request by Harikesh00**
Added test cases for `DispatchRequestOptions`, `TransferRequest`, `SignaturesProtectedBranch`, `DismissalRestrictionsRequest` of file `github/repos.go`

contribution to #55

### CODE REVIEW

File: github/repos_test.go [hunk 1]
```diff
@@ -3002,3 +3002,80 @@ func TestAuthorizedActorsOnly_Marshal(t *testing.T) {
...
+func TestDispatchRequestOptions_Marshal(t *testing.T) {
+	testJSONMarshal(t, &DispatchRequestOptions{}, "{}")
+
+	u := &DispatchRequestOptions{
+		EventType: "test_event_type",
+	}
+
+	want := `{
+		"event_type":"test_event_type",
+		"client_payload":null
```
Comment (gmlewis) [inline github/repos_test.go:3015]: Can you simulate a client payload here?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Harikesh00 !

**Comment by gmlewis** [comment]
@Harikesh00 - do you want Hacktoberfest credit for this one?  If so, please add to the description in future PRs. Thank you!

File: github/repos_test.go (hunk 1 above)
Comment (Harikesh00) [reply github/repos_test.go:3015]: sure, I'll do & push the changes

**Comment by Harikesh00** [comment]
yes @gmlewis. I want this for Hacktoberfest, kindly add the label for same

File: github/repos_test.go (hunk 1 above)
Comment (Harikesh00) [reply github/repos_test.go]: changes pushed

**Review by gmlewis — CHANGES_REQUESTED** [review]
Just for consistency sake, could you please add in the whitespace to all the `want` strings that the standard Go `json.MarshalIndent` would add so that all our JSON examples look the same?

**Comment by Harikesh00** [comment]
spaces added
plz check

**Pull request merged**